tests/test_dynamics_bookmarks.py

```diff
@@ -0,0 +1,213 @@
+import datetime
+import dateutil.parser
+import pytz
+
+from tap_tester import runner, menagerie, connections
+
+from base import DynamicsBaseTest
+
+
+class DynamicsBookmarks(DynamicsBaseTest):
+    @staticmethod
+    def name():
+        return "tap_tester_dynamics_bookmarks"
+
+    def get_properties(self, original: bool = True):
+        """Configuration properties required for the tap."""
+        return_value = {
+            "start_date": "2021-04-01T00:00:00Z",
+            "user_agent": "tap-dynamics <api_user_email@your_company.com>",
+        }
+        if original:
+            return return_value
+
+        return_value["start_date"] = self.start_date
+
+        return return_value
+
+    @staticmethod
+    def convert_state_to_utc(date_str):
+        """
+        Convert a saved bookmark value of the form '2020-08-25T13:17:36-07:00' to
+        a string formatted utc datetime,
+        in order to compare against json formatted datetime values
+        """
+        date_object = dateutil.parser.parse(date_str)
+        date_object_utc = date_object.astimezone(tz=pytz.UTC)
+        return datetime.datetime.strftime(date_object_utc, "%Y-%m-%dT%H:%M:%SZ")
+
+    def calculated_states_by_stream(self, current_state):
+        """
+        Look at the bookmarks from a previous sync and set a new bookmark
+        value based off timedelta expectations. This ensures the subsequent sync will replicate
+        at least 1 record but, fewer records than the previous sync.
+
+        Sufficient test data is required for this test to cover a given stream.
+        An incremental replication stream must have at least two records with
+        replication keys that differ by more than the lookback window.
+
+        If the test data is changed in the future this will break expectations for this test.
+
+        The following streams barely make the cut:
+        account         "2021-02-09T18:17:30.000000Z"
```

Comment:
I'm not sure if the docstring is accurate here. It looks like these dates are prior to the start_date and would not be replicated. Also it says there is a lookback window, which I don't believe is true.